pkg/clusterresource/openstack.go

```diff
@@ -43,6 +43,17 @@ type OpenStackCloudBuilder struct {
 	MasterFlavor string
 }
 
+func NewOpenStackCloudBuilderFromSecret(credsSecret *corev1.Secret) *OpenStackCloudBuilder {
+	cloudsYamlContent := credsSecret.Data[constants.OpenStackCredentialsName]
+	return &OpenStackCloudBuilder{
+		CloudsYAMLContent: cloudsYamlContent,
+		Cloud:             "openstack",
+		ComputeFlavor:     "m1.large",
+		MasterFlavor:      "ci.m4.xlarge",
+		ExternalNetwork:   "provider_net_shared_3",
```

Comment:
This all looks very much hard-coded for what I believe is intended to be generic support for OpenStack.

Reply:
Default values - it exists in other providers (aws, azure, gcp) but using const and a bit different stracture, but non the less hardcoded. I can align with the other providers stracture though.

Reply:
if you look at the usage of this function in other clouds (aws, azure, gcp), the function intends to create CloudBuilder from the contents of the creds secret. Some of the constants here are not fetched from secret. I think these values can be fetched from the install config instead.

Reply:
The values don't exist in install config when user doesn't provide a template. I will update how the default values are injected to match the other clouds.